app/controllers/api/v1x0/requests_controller.rb

```diff
@@ -25,6 +25,19 @@ def index
         collection(reqs)
       end
 
+      def cancel
+        req = Request.find(params.require(:request_id))
+        raise Exceptions::ApprovalError, "Unable to cancel request." unless req.current_stage
```

Comment:
This naming seems odd. It reads like "Do this unless the current stage is....?" Perhaps `req.past_cancellation?` or `req.processing?` or something similar would be better.

Reply:
`current_stage` really means the current processing stage of the request. It will be used later to create action on it. Request is finished when `current_stage is nil`.

Reply:
Yeah I understand what it means, but it just reads strangely because you would expect that there are some stages that are unable to be canceled. Right now it looks like as long as there is a current stage, it's cancelable (which might be true, but again, it just reads strangely).

Also, since you are using it below in the action create, you should probably localize it in a variable so you're not hitting the database both times you call `.current_stage`.

Reply:
We can introduce a method like `cancelable?` in request. Localizing is not important here. It does not hit the database multiple times.

Reply:
I realize this is probably going to change, but `current_stage` right now calls `find_by`, and then again on line 32 we're calling `req.current_stage.id`. Does this not call `find_by` twice?

Reply:
@bzwei After endpoint changes to `requests/:id/actions`, it's only limit to `cancel` only. So `cancelable?` is unsuitable here. How about  `actionable?`